pkg/api/platform/v1/decode.go

```diff
@@ -166,6 +193,105 @@ func initialise(msg proto.Message) {
 			}
 			initialise(reflectMsg.Get(field).Message().Interface())
 		}
+	}
+}
+
+func cleanProto(msg proto.Message) {
+	if msg == nil {
+		return
+	}
+	cleanReflect(msg.ProtoReflect())
+}
+
+func cleanReflect(msg protoreflect.Message) {
+	if msg == nil {
+		return
+	}
+
+	fields := msg.Descriptor().Fields()
+	for i := 0; i < fields.Len(); i++ {
+		field := fields.Get(i)
+		value := msg.Get(field)
+		if field.IsMap() {
```

Comment:
You will want to visit the value of lists and maps (recursive call to cleanReflect of list and map Value)?

Reply:
And be sure to not remove them if they are empty in that immediate case :sweat_smile:

Reply:
Just realized that you only remove sub-fields, so disregard my last message :D But for the msg case, I wonder if you should do the empty-check after the recursive visit in case it trims elements (not 100% convinced if the empty-check matches the behavior or empty maps/lists - though it should, right?)?

Reply:
Yea that would make sense to do